pkg/morph/subscriber/subscriber.go

```diff
@@ -199,16 +163,161 @@ func New(ctx context.Context, p *Params) (Subscriber, error) {
 		notifyChan: make(chan *state.ContainedNotificationEvent),
 		blockChan:  make(chan *block.Block),
 		notaryChan: make(chan *result.NotaryRequestEvent),
-	}
 
-	// Worker listens all events from neo-go websocket and puts them
-	// into corresponding channel. It may be notifications, transactions,
-	// new blocks. For now only notifications.
+		curNotifyChan: make(chan *state.ContainedNotificationEvent),
+		curBlockChan:  make(chan *block.Block),
+		curNotaryChan: make(chan *result.NotaryRequestEvent),
+
+		subscribedEvents:       make(map[util.Uint160]bool),
+		subscribedNotaryEvents: make(map[util.Uint160]bool),
+	}
+	// Worker listens all events from temporary NeoGo channel and puts them
+	// into corresponding permanent channels.
 	go sub.routeNotifications(ctx)
 
 	return sub, nil
 }
 
+func (s *subscriber) routeNotifications(ctx context.Context) {
+	var (
+		// TODO: not needed after nspcc-dev/neo-go#2980.
+		cliCh             = s.client.NotificationChannel()
```

Comment:
We may safely remove this old shared channel now.

Reply:
Then we need to add `no-lint` comment to the usage of deprecated WSClient's shared channel.